tests/phpunit/tests/theme/wpThemeJsonResolver.php

```diff
@@ -365,12 +369,56 @@ function test_get_user_data_from_wp_global_styles_does_not_use_uncached_queries(
 		$this->assertNotEmpty( $user_cpt );
 
 		$query_count = count( $this->queries );
-		for ( $i = 0; $i < 3; $i++ ) {
-			WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
+		for ( $i = 0; $i < 3; $i ++ ) {
+			$new_user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme() );
 			WP_Theme_JSON_Resolver::clean_cached_data();
+			$this->assertSameSets( $user_cpt, $new_user_cpt );
 		}
 		$query_count = count( $this->queries ) - $query_count;
 		$this->assertEquals( 0, $query_count, 'Unexpected SQL queries detected for the wp_global_style post type' );
 		remove_filter( 'query', array( $this, 'filter_db_query' ) );
 	}
+
+	/**
+	 * @covers WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles
+	 * @ticket 55392
+	 */
+	function test_get_user_data_from_wp_global_styles_does_exist() {
+		$post1 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme(), true );
+		$this->assertIsArray( $post1 );
+		$this->assertArrayHasKey( 'ID', $post1 );
+		wp_delete_post( $post1['ID'], true );
+		$post2 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme(), true );
+		$this->assertIsArray( $post2 );
+		$this->assertArrayHasKey( 'ID', $post2 );
+	}
+
+	/**
+	 * @covers WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles
+	 * @ticket 55392
+	 */
+	function test_get_user_data_from_wp_global_styles_create_post() {
+		$post1 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme( 'testing' ) );
+		$this->assertIsArray( $post1 );
+		$this->assertSameSets( array(), $post1 );
+		$post2 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme( 'testing' ) );
+		$this->assertIsArray( $post2 );
+		$this->assertSameSets( array(), $post2 );
+		$post3 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( wp_get_theme( 'testing' ), true );
```

Comment:
Can we set one variable for `wp_get_theme( 'testing' )` and use it?

i.e.
```suggestion
		$theme = wp_get_theme( 'testing' );
		$post1 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme );
		$this->assertIsArray( $post1 );
		$this->assertSameSets( array(), $post1 );
		$post2 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme );
		$this->assertIsArray( $post2 );
		$this->assertSameSets( array(), $post2 );
		$post3 = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme, true );
```

This also applies to the other test method.